app/models/gallery.rb

```diff
@@ -75,12 +76,15 @@ def to_param
   def set_images_from_portal_urls
     transaction do
       enumerable_image_portal_urls.map { |url| Europeana::Record.id_from_portal_url(url) }.tap do |record_ids|
+        matched_ids = []
         record_ids.each_with_index do |record_id, i|
-          GalleryImage.find_or_create_by(gallery: self, europeana_record_id: record_id).tap do |image|
+          image_url = response_items.detect { |record| record['id'] == record_id }['edmIsShownBy'].first
+          GalleryImage.find_or_create_by!(gallery: self, europeana_record_id: record_id, image_url: image_url).tap do |image|
```

Comment:
Should this really be `GalleryImage.find_or_create_by!` instead of `GalleryImage.find_or_create_by`? Won't the former prevent display of errors for all invalid image URLs at once?